src/SDK/SmartDetectorsSDK/AnalysisRequest.cs

```diff
@@ -23,10 +22,17 @@ public struct AnalysisRequest : IEquatable<AnalysisRequest>
         /// </summary>
         /// <param name="targetResources">The list of resource identifiers to analyze.</param>
         /// <param name="analysisCadence">The analysis cadence defined in the Alert Rule which initiated the Smart Detector's analysis.</param>
-        /// <param name="alertRuleResourceId">The alert rule resource ID.</param>
+        /// <param name="alertRuleResourceId">The Alert Rule resource ID.</param>
+        /// <param name="detectorParameters">The detector parameters as specified in the Alert Rule.</param>
         /// <param name="analysisServicesFactory">The analysis services factory to be used for querying the resources telemetry.</param>
         /// <param name="stateRepository">The persistent state repository for storing state between analysis runs</param>
-        public AnalysisRequest(List<ResourceIdentifier> targetResources, TimeSpan analysisCadence, string alertRuleResourceId, IAnalysisServicesFactory analysisServicesFactory, IStateRepository stateRepository)
+        public AnalysisRequest(
+            List<ResourceIdentifier> targetResources,
+            TimeSpan analysisCadence,
+            string alertRuleResourceId,
+            IDictionary<string, object> detectorParameters,
```

Comment:
Consider also doing it IReadOnlyDictionary

Reply:
I can't, because you didn't push the nuget with the read only version

---
In reply to: [233431827](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/23#discussion_r233431827) [](ancestors = 233431827)

Reply:
OMG

Reply:
??

---
In reply to: [233641479](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/23#discussion_r233641479) [](ancestors = 233641479)